tests/test_policy.py

```diff
@@ -500,6 +500,66 @@ def test_resource_type_empty_metadata(self):
         if empty:
             raise ValueError("Empty Resource Metadata %s" % (', '.join(empty)))
 
+    def test_valid_arn_type(self):
+        arn_db = load_data('arn-types.json')
+        invalid = {}
+        overrides = {'wafv2': set(('webacl',))}
+
+        # we have a few resources where we have synthetic arns
+        # or they aren't in the iam ref docs.
+        allow_list = set((
+            # bug in the arnref script or test logic below.
+            'glue-catalog',
+            # these are valid, but v1 & v2 arns get mangled into the
+            # same top level prefix
+            'rest-api',
+            'rest-stage',
+            # synthetics ~ ie. c7n introduced since non exist.
+            # or in some cases where it exists but not usable in iam.
+            'scaling-policy',
+            'glue-classifier',
+            'glue-security-configuration',
+            'event-rule-target',
+            'rrset',
+            'redshift-reserved',
+            'elasticsearch-reserved'
+        ))
+
+        for k, v in manager.resources.items():
+            if k in allow_list:
+                continue
+            svc = v.resource_type.service
+            if not v.resource_type.arn_type:
+                continue
+
+            svc_arn_map = arn_db.get(svc, {})
+            if not svc_arn_map:
+                continue
+
+            svc_arns = list(svc_arn_map.values())
+            svc_arn_types = set()
+            for sa in svc_arns:
```

Comment:
Isn't think going over every arn in a service for every resource type in that service?

Reply:
yes but service name for a resource type isn't the same as custodian, what we're generally validating is that custodian is using a valid arn resource type, not direct match to service resource type which requires additional extraneous mappings that aren't needed for runtime.